src/resources/Charges.ts

```diff
@@ -130,29 +130,15 @@ export class Charges extends CRUDResource {
     poll(
         storeId: string,
         id: string,
-        data?: SendData<PollParams>,
+        data?: SendData<PollData>,
         auth?: AuthParams,
         callback?: ResponseCallback<ResponseCharge>,
-
-        /**
-         * Condition for the resource to be successfully loaded. Default to pending status check.
-         */
-        cancelCondition?: (response: ResponseCharge) => boolean,
-        successCondition: ({ status }: ResponseCharge) => boolean = ({ status }) => status !== ChargeStatus.PENDING,
-        iterationCallback?: (response: ResponseCharge) => void,
-        pollOptions?: { interval?: number; timeout?: number }
+        pollParams?: Partial<PollParams<ResponseCharge>>
```

Comment:
hm... `pollParams` coming the `callback` is somewhat iffy... I'm almost tempted to say, let's get rid of the callbacks...

Reply:
I tried to keep parameters common to all requests together.

I am also tempted to remove the callback and let people use `.then` to be honest...

Reply:
Or maybe should we use the same for `auth` and `callback`? Like `requestParams`? 🤔

Reply:
If we make a breaking release I think remove all callbacks from the SDKs...